src/metabase/api/pulse.clj

```diff
@@ -33,9 +33,11 @@
                   can-write?)]
     (assoc pulse :read_only (not can-write?))))
 
-
-(defn- check-card-read-permissions [cards]
-  (doseq [{card-id :id} cards]
+(defn check-card-read-permissions
+  "Users can only create a pulse for `CARDS` they have access to"
+  [cards]
+  (doseq [card cards
+          :let [card-id (u/get-id card)]]
     (assert (integer? card-id))
```

Comment:
`u/get-id` already asserts that the output is a valid Integer ID so you don't need to do another `assert` below it

That's sort of the idea behind `u/get-id`: it's guaranteed to return a valid ID. Oh yeah that and it can accept either a map with an `:id` key or an integer which lets you write function that take args like `object-or-id` and handle either case

Reply:
At any rate you should just be able to do `(api/read-check card)` here instead of `(api/read-check Card (u/get-id card))`. 

The latter essentially just translates to `(api/read-check (db/select-one Card :id <id>))` so this will end up doing a DB fetch for each Card. Since you already have the cards those fetches can and should be skipped